lib/assembly-objectfile/object_fileable.rb

```diff
@@ -110,22 +114,33 @@ def sha1
       @sha1 ||= Digest::SHA1.file(path).hexdigest
     end
 
-    # Returns mimetype information for the current file based on
-    #   (1) exifdata (if available), (2) unix file type or (3) file extension using the mimetypes gem, in this priority order
+    # Returns mimetype information for the current file based on the ordering set in default_mime_type_order
+    #   We stop computing mimetypes as soon as we have a method that returns a value
     # @return [String] mime type
     # @example
     #   source_file = Assembly::ObjectFile.new('/input/path_to_file.txt')
     #   puts source_file.mimetype # 'text/plain'
     def mimetype
       @mimetype ||= begin
-        if exif_mimetype # first, try the exif data
-          exif_mimetype
-        elsif file_mimetype # next, try exif/unix file system command
-          file_mimetype
-        else # finally, get it from the mime-types gem (using the file extension) if both of those failed for some reason
-          mtype = MIME::Types.type_for(path).first
-          mtype ? mtype.content_type : ''
+        check_for_file
+        mimetype = ''
+        mime_type_order.each do |mime_type_method|
+          mimetype = public_send("#{mime_type_method}_mimetype") if VALID_MIMETYPE_METHODS.include?(mime_type_method)
+          break if (!mimetype.nil? && mimetype != '')
```

Comment:
stop generating mimetypes when we hit the first one in our specified order that isn't nil or blank